### PR TITLE
feat: masa stack config  support dcc

### PR DESCRIPTION
**Pull request by codding-y**

### CODE REVIEW

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/MasaStackConfigExtensions.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ public static string GetPmServiceDomain(this IMasaStackConfig masaStackConfig)
 
     public static string GetDccServiceDomain(this IMasaStackConfig masaStackConfig)
     {
-        return GetServerDomain(masaStackConfig, HttpProtocol.HTTP, MasaStackConstant.DCC, MasaStackConstant.SERVER);
+        return GetServerDomain(masaStackConfig, HttpProtocol.HTTPS, MasaStackConstant.DCC, MasaStackConstant.SERVER);
```
Comment (MayueCif) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/MasaStackConfigExtensions.cs:93]: ？

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/MasaStackConfigOptions.cs [hunk 2]
```diff
@@ -5,15 +5,23 @@ namespace Masa.Contrib.StackSdks.Config;
 
 public class MasaStackConfigOptions
```
Comment (MayueCif) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/MasaStackConfigOptions.cs:6]: static

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/ServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -5,43 +5,98 @@ namespace Microsoft.Extensions.DependencyInjection;
...
+    public static IServiceCollection AddMasaStackConfig(this IServiceCollection services, bool init = false)
     {
-        services.TryAddSingleton<IMasaStackConfig, MasaStackConfig>();
         if (init)
         {
             InitializeMasaStackConfiguration(services);
         }
+        else
+        {
+            services.TryAddScoped<IMasaStackConfig>(serviceProvider =>
```
Comment (MayueCif) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/ServiceCollectionExtensions.cs:80]: single

**Review by MayueCif — APPROVED** [review]
@zhenlei520

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/MasaStackConfigExtensions.cs [hunk 4]
```diff
@@ -169,8 +169,9 @@ public static T GetDccMiniOptions<T>(this IMasaStackConfig masaStackConfig)
         stringBuilder.Append($"\"{masaStackConfig.DccSecret}\",");
         stringBuilder.Append(@"""PublicSecret"":");
         stringBuilder.Append($"\"{masaStackConfig.DccSecret}\"");
-        stringBuilder.Append(@"}");
-        return JsonSerializer.Deserialize<T>(stringBuilder.ToString()) ?? throw new JsonException();
+        stringBuilder.Append('}');
+
+        return JsonSerializer.Deserialize<DccOptions>(stringBuilder.ToString()) ?? throw new JsonException();
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/MasaStackConfigExtensions.cs:174]: Why not manually initialize DccOptions and assign values?

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/ServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -5,43 +5,101 @@ namespace Microsoft.Extensions.DependencyInjection;
...
+        {
+            var remoteConfigs = await configurationApiClient.GetAsync<Dictionary<string, string>>(
+                masaStackConfig.Environment,
+                masaStackConfig.Cluster,
+                DEFAULT_PUBLIC_ID,
+                DEFAULT_CONFIG_NAME,
+                value =>
+                {
+                    masaStackConfig.UpdateMasaStackConfigContent(value);
+                });
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/ServiceCollectionExtensions.cs:54]: After the remote Dcc configuration is changed, it is inappropriate to change the obtained configuration immediately. In the same request, the obtained configuration should always be the same, even if the configuration is modified before the request ends

File: src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/ServiceCollectionExtensions.cs [hunk 6]
```diff
@@ -5,43 +5,101 @@ namespace Microsoft.Extensions.DependencyInjection;
...
         {
-            InitializeMasaStackConfiguration(services);
+            InitializeMasaStackConfiguration(services).ConfigureAwait(false);
+        }
+        else
+        {
+            services.TryAddScoped<IMasaStackConfig>(serviceProvider =>
+            {
+                var client = serviceProvider.GetRequiredService<IConfigurationApiClient>();
+                return new MasaStackConfig(new(), client);
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Masa.Contrib.StackSdks.Config/ServiceCollectionExtensions.cs:86]: Use a constructor with only one parameter of type IConfigurationApiClient

File: src/Contrib/StackSdks/Tests/Masa.Contrib.StackSdks.Middleware.Tests/RequestMiddlewareTest.cs [hunk 7]
```diff
@@ -1,6 +1,10 @@
 // Copyright (c) MASA Stack All rights reserved.
 // Licensed under the MIT License. See LICENSE.txt in the project root for license information.
 
+using Masa.BuildingBlocks.StackSdks.Config;
```
Comment (zhenlei520) [inline src/Contrib/StackSdks/Tests/Masa.Contrib.StackSdks.Middleware.Tests/RequestMiddlewareTest.cs:4]: use global using

**Review by zhenlei520 — CHANGES_REQUESTED** [review]

**Review by zhenlei520 — APPROVED** [review]

**Pull request merged**